packages/components/src/Filter/Filter.tsx

```diff
@@ -8,11 +8,12 @@ export interface FilterProps
   extends OverrideClassName<HTMLAttributes<HTMLDivElement>> {
   children: React.ReactNode
   isOpen: boolean
-  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>
+  setIsOpen: (isOpen: boolean) => void
```

Comment:
Why do we need this prop? Wouldn't it be possible to achieve the same affect toggling `isOpen`?

Reply:
The prop exists because we expect the consumer to hold the state for `isOpen`. The toggle functionality is used because we pass them to the trigger so the consumer doesn't need to do it o3o

Reply:
Wouldn't this then be more accurately named `handleOnClick`?

Reply:
`onTriggerClick`? :P

The problem with that is it isn't obvious that I need the open state handler in there. Then you'll just end up with something that appears borked. Not to mention, because it's passed back to the trigger as props, you can actually extend it as the consumer through there, and not put it into the `Filter` prop.